src/compile.rs

```diff
@@ -209,20 +193,45 @@ fn compile_target(
         }
     }
 
-    let cargo_args = vec!["rustc", "--message-format", "json"];
+    let cargo_args = vec!["--message-format", "json"];
 
     let build_args: Vec<_> = cargo_args
         .iter()
         .chain(&shared_args)
         .chain(&["--"])
         .chain(&rustc_args)
         .collect();
-    let command_str = build_args
-        .iter()
-        .fold("cargo".to_string(), |acc, x| acc + " " + x);
-
-    let mut let_binding = Command::new("cargo");
-    let build_command = let_binding
+    let command_str = build_args.iter().fold(
```

Comment:
you can [join](https://doc.rust-lang.org/std/primitive.slice.html#method.join) here, fold is inscrutable to someone without functional programming background